modules/codegen/src/main/scala/com/twilio/guardrail/generators/circe/protocol/CirceProtocolGenerator.scala

```diff
@@ -313,45 +324,59 @@ object CirceProtocolGenerator {
   object PolyProtocolTermInterp extends (PolyProtocolTerm ~> Target) {
     override def apply[A](fa: PolyProtocolTerm[A]): Target[A] = fa match {
       case RenderADTCompanion(clsName, discriminator, encoder, decoder) =>
-        val code = q"""object ${Term.Name(clsName)} extends AutoDerivation {
-            ..${List(discriminator, encoder, decoder)}
+        val code = q"""object ${Term.Name(clsName)} {
+             val discriminator:String = ${Lit.String(discriminator)}
+            ..${List(encoder, decoder)}
           }
           """
         Target.pure(code)
 
-      case RenderDiscriminator(discriminator) =>
-        val code = q"""implicit val configuration: Configuration = Configuration.default.withDiscriminator($discriminator)"""
-        Target.pure(code)
-
-      case DecodeADT(clsName, needCamelSnakeConversion) =>
-        val code = if (needCamelSnakeConversion) {
-          q"""implicit val decoder: Decoder[${Type.Name(clsName)}] =
-                  deriveDecoder[${Type.Name(clsName)}](io.circe.derivation.renaming.snakeCase)"""
-        } else {
-          q"""implicit val decoder: Decoder[${Type.Name(clsName)}] = deriveDecoder[${Type.Name(clsName)}]"""
-        }
+      case DecodeADT(clsName, children) =>
+        val childrenCases = children.map(
+          child => p"case ${Lit.String(child)} => c.as[${Type.Name(child)}]"
```

Comment:
This requires the child types have the same name as the discriminator values? Is there any chance of these getting out of sync or conflicting in some way?

Reply:
Currently discriminator/mapping doesn't support. Of course it's make sense to support that feature, but could I make it later?)

Reply:
I'm thinking more about using a path for the type instead of a bare identifier.

Consider if my discriminator is `Either` or `Int` or something like that.

Better to use a structure like `AbstractPet.Dog` to avoid this entirely, as it is impossible for user input to cause a failure.

Reply:
I refer to the swagger's documentation. And there is no word about that(
https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/